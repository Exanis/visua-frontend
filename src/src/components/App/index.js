import React from 'react';
import configureStore from '../../store';
import {connect} from 'react-redux';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router';
import { Route, Switch } from 'react-router';
import Layout from '../Layout';
import createHistory from 'history/createBrowserHistory';

import Login from "../Pages/Login";
import Profile from "../Pages/Profile";
import Users from '../Pages/Users';
import Pipeline from '../Pages/Pipeline';
import Block from '../Pages/Block';
import Runner from '../Pages/Runner';
import Editor from '../Pages/Editor';

const language =
  (navigator.languages && navigator.languages[0]) ||
  navigator.language ||
  navigator.userLanguage;

const languageWithoutRegionCode = language.toLowerCase().split(/[_-]+/)[0];

const history = createHistory();
const store = configureStore(history);

class PrivateRouteComponent extends React.Component {
    render () {
        const { component, ...others } = this.props;

        if (!this.props.user.anonymous) {
            return <Route {...this.props} />;
        }
        return <Route component={Login} {...others} />;
    }
}

class StaffRouteComponent extends React.Component {
    render () {
        const { component, ...others } = this.props;

        if (!this.props.user.anonymous && this.props.user.info.is_staff) {
            return <Route {...this.props} />;
        }
        return <Route component={Login} {...others} />;
    }
}

const mapStateToProps = state => {
    return {
        user: state.user
    };
};

const PrivateRoute = connect(mapStateToProps)(PrivateRouteComponent);
const StaffRoute = connect(mapStateToProps)(StaffRouteComponent);

class App extends React.Component {
    render () {
        return <Provider store={store}>
            <IntlProvider locale={languageWithoutRegionCode}>
                <ConnectedRouter history={history}>
                    <Layout>
                        <Switch>
                            <Route exact path={'/'} component={Login} />

                            <PrivateRoute exact path={'/home'} component={Pipeline} />
                            <PrivateRoute path={'/editor/:pipelineId'} component={Editor} />
                            <PrivateRoute exact path={'/profile'} component={Profile} />

                            <StaffRoute exact path={'/users'} component={Users} />
                            <StaffRoute exact path={'/block'} component={Block} />
                            <StaffRoute exact path={'/runner'} component={Runner} />
                        </Switch>
                    </Layout>
                </ConnectedRouter>
            </IntlProvider>
        </Provider>;
    }
}

App.propTypes = {};

export default App;