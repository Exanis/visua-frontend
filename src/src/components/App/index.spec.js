import React from 'react';
import { shallow } from 'enzyme';

import App from './index';

describe('<App />', () => {
    it('should render App properly', () => {
        const props = {
            classes: {},
            requestCurrentUserData: () => {}
        };
        const rendererComponent = shallow(<App {...props} />)

        expect(rendererComponent).toMatchSnapshot();
    });
});