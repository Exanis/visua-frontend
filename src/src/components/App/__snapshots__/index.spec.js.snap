// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    classes={Object {}}
    requestCurrentUserData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <IntlProvider
        locale="en"
>
        <Connect(ConnectedRouter)
                history={
                        Object {
                                "action": "POP",
                                "block": [Function],
                                "createHref": [Function],
                                "go": [Function],
                                "goBack": [Function],
                                "goForward": [Function],
                                "length": 1,
                                "listen": [Function],
                                "location": Object {
                                  "hash": "",
                                  "pathname": "/",
                                  "search": "",
                                  "state": undefined,
                                },
                                "push": [Function],
                                "replace": [Function],
                              }
                }
        >
                <withRouter(Connect(WithStyles(Layout)))>
                        <Switch>
                                <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                                />
                                <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/home"
                                />
                                <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        path="/editor/:pipelineId"
                                />
                                <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/profile"
                                />
                                <Connect(StaffRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/users"
                                />
                                <Connect(StaffRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/block"
                                />
                        </Switch>
                </withRouter(Connect(WithStyles(Layout)))>
        </Connect(ConnectedRouter)>
</IntlProvider>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(ConnectedRouter)
          history={
                    Object {
                              "action": "POP",
                              "block": [Function],
                              "createHref": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "length": 1,
                              "listen": [Function],
                              "location": Object {
                                "hash": "",
                                "pathname": "/",
                                "search": "",
                                "state": undefined,
                              },
                              "push": [Function],
                              "replace": [Function],
                            }
          }
>
          <withRouter(Connect(WithStyles(Layout)))>
                    <Switch>
                              <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/"
                              />
                              <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/home"
                              />
                              <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        path="/editor/:pipelineId"
                              />
                              <Connect(PrivateRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/profile"
                              />
                              <Connect(StaffRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/users"
                              />
                              <Connect(StaffRouteComponent)
                                        component={[Function]}
                                        exact={true}
                                        path="/block"
                              />
                    </Switch>
          </withRouter(Connect(WithStyles(Layout)))>
</Connect(ConnectedRouter)>,
        "locale": "en",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <withRouter(Connect(WithStyles(Layout)))>
            <Switch>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <Connect(PrivateRouteComponent)
                                    component={[Function]}
                                    exact={true}
                                    path="/home"
                        />
                        <Connect(PrivateRouteComponent)
                                    component={[Function]}
                                    path="/editor/:pipelineId"
                        />
                        <Connect(PrivateRouteComponent)
                                    component={[Function]}
                                    exact={true}
                                    path="/profile"
                        />
                        <Connect(StaffRouteComponent)
                                    component={[Function]}
                                    exact={true}
                                    path="/users"
                        />
                        <Connect(StaffRouteComponent)
                                    component={[Function]}
                                    exact={true}
                                    path="/block"
                        />
            </Switch>
</withRouter(Connect(WithStyles(Layout)))>,
          "history": Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Connect(PrivateRouteComponent)
                            component={[Function]}
                            exact={true}
                            path="/home"
              />
              <Connect(PrivateRouteComponent)
                            component={[Function]}
                            path="/editor/:pipelineId"
              />
              <Connect(PrivateRouteComponent)
                            component={[Function]}
                            exact={true}
                            path="/profile"
              />
              <Connect(StaffRouteComponent)
                            component={[Function]}
                            exact={true}
                            path="/users"
              />
              <Connect(StaffRouteComponent)
                            component={[Function]}
                            exact={true}
                            path="/block"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Connect(PrivateRouteComponent)
                  component={[Function]}
                  exact={true}
                  path="/home"
/>,
                <Connect(PrivateRouteComponent)
                  component={[Function]}
                  path="/editor/:pipelineId"
/>,
                <Connect(PrivateRouteComponent)
                  component={[Function]}
                  exact={true}
                  path="/profile"
/>,
                <Connect(StaffRouteComponent)
                  component={[Function]}
                  exact={true}
                  path="/users"
/>,
                <Connect(StaffRouteComponent)
                  component={[Function]}
                  exact={true}
                  path="/block"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/home",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/editor/:pipelineId",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/profile",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/users",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/block",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <IntlProvider
          locale="en"
>
          <Connect(ConnectedRouter)
                    history={
                              Object {
                                        "action": "POP",
                                        "block": [Function],
                                        "createHref": [Function],
                                        "go": [Function],
                                        "goBack": [Function],
                                        "goForward": [Function],
                                        "length": 1,
                                        "listen": [Function],
                                        "location": Object {
                                          "hash": "",
                                          "pathname": "/",
                                          "search": "",
                                          "state": undefined,
                                        },
                                        "push": [Function],
                                        "replace": [Function],
                                      }
                    }
          >
                    <withRouter(Connect(WithStyles(Layout)))>
                              <Switch>
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/"
                                        />
                                        <Connect(PrivateRouteComponent)
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/home"
                                        />
                                        <Connect(PrivateRouteComponent)
                                                  component={[Function]}
                                                  path="/editor/:pipelineId"
                                        />
                                        <Connect(PrivateRouteComponent)
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/profile"
                                        />
                                        <Connect(StaffRouteComponent)
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/users"
                                        />
                                        <Connect(StaffRouteComponent)
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/block"
                                        />
                              </Switch>
                    </withRouter(Connect(WithStyles(Layout)))>
          </Connect(ConnectedRouter)>
</IntlProvider>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(ConnectedRouter)
            history={
                        Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "pathname": "/",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
            }
>
            <withRouter(Connect(WithStyles(Layout)))>
                        <Switch>
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/"
                                    />
                                    <Connect(PrivateRouteComponent)
                                                component={[Function]}
                                                exact={true}
                                                path="/home"
                                    />
                                    <Connect(PrivateRouteComponent)
                                                component={[Function]}
                                                path="/editor/:pipelineId"
                                    />
                                    <Connect(PrivateRouteComponent)
                                                component={[Function]}
                                                exact={true}
                                                path="/profile"
                                    />
                                    <Connect(StaffRouteComponent)
                                                component={[Function]}
                                                exact={true}
                                                path="/users"
                                    />
                                    <Connect(StaffRouteComponent)
                                                component={[Function]}
                                                exact={true}
                                                path="/block"
                                    />
                        </Switch>
            </withRouter(Connect(WithStyles(Layout)))>
</Connect(ConnectedRouter)>,
          "locale": "en",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <withRouter(Connect(WithStyles(Layout)))>
              <Switch>
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/"
                            />
                            <Connect(PrivateRouteComponent)
                                          component={[Function]}
                                          exact={true}
                                          path="/home"
                            />
                            <Connect(PrivateRouteComponent)
                                          component={[Function]}
                                          path="/editor/:pipelineId"
                            />
                            <Connect(PrivateRouteComponent)
                                          component={[Function]}
                                          exact={true}
                                          path="/profile"
                            />
                            <Connect(StaffRouteComponent)
                                          component={[Function]}
                                          exact={true}
                                          path="/users"
                            />
                            <Connect(StaffRouteComponent)
                                          component={[Function]}
                                          exact={true}
                                          path="/block"
                            />
              </Switch>
</withRouter(Connect(WithStyles(Layout)))>,
            "history": Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Switch>
                <Route
                                component={[Function]}
                                exact={true}
                                path="/"
                />
                <Connect(PrivateRouteComponent)
                                component={[Function]}
                                exact={true}
                                path="/home"
                />
                <Connect(PrivateRouteComponent)
                                component={[Function]}
                                path="/editor/:pipelineId"
                />
                <Connect(PrivateRouteComponent)
                                component={[Function]}
                                exact={true}
                                path="/profile"
                />
                <Connect(StaffRouteComponent)
                                component={[Function]}
                                exact={true}
                                path="/users"
                />
                <Connect(StaffRouteComponent)
                                component={[Function]}
                                exact={true}
                                path="/block"
                />
</Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
/>,
                  <Connect(PrivateRouteComponent)
                    component={[Function]}
                    exact={true}
                    path="/home"
/>,
                  <Connect(PrivateRouteComponent)
                    component={[Function]}
                    path="/editor/:pipelineId"
/>,
                  <Connect(PrivateRouteComponent)
                    component={[Function]}
                    exact={true}
                    path="/profile"
/>,
                  <Connect(StaffRouteComponent)
                    component={[Function]}
                    exact={true}
                    path="/users"
/>,
                  <Connect(StaffRouteComponent)
                    component={[Function]}
                    exact={true}
                    path="/block"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/home",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/editor/:pipelineId",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/profile",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/users",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/block",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
